lib/gooddata/models/data_source.rb

```diff
@@ -0,0 +1,648 @@
+# encoding: UTF-8
+# frozen_string_literal: true
+#
+# Copyright (c) 2010-2021 GoodData Corporation. All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+require_relative '../rest/resource'
+
+module GoodData
+  class DataSource < Rest::Resource
+    attr_accessor :connection_info
+
+    DATA_SOURCES_URL = '/gdc/dataload/dataSources'
+    SNOWFLAKE = 'snowflake'
+    REDSHIFT = 'redshift'
+    BIGQUERY = 'bigQuery'
+    GENERIC = 'generic'
+    S3 = 's3'
+    ADS = 'ads'
+
+    class << self
+      # Get all data sources or get a specify data source from data source identify
+      # Expected parameter value:
+      # - :all return all data sources
+      # - :data_source_id return a data source with the specify data source identify
+      def [](id = :all, options = { client: GoodData.client })
+        c = GoodData.get_client(options)
+
+        if id == :all
+          data = c.get(DATA_SOURCES_URL)
+          data['dataSources']['items'].map do |ds_data|
+            c.create(DataSource, ds_data)
+          end
+        else
+          c.create(DataSource, c.get("#{DATA_SOURCES_URL}/#{id}"))
+        end
+      end
+
+      # Get a specify data source from data source identify
+      def from_id(id, options = { client: GoodData.client })
+        DataSource[id, options]
+      end
+
+      # Get a specify data source from data source alias
+      def from_alias(data_source_alias, options = { client: GoodData.client })
+        data_sources = all(options)
+        data_sources.find do |data_source|
+          data_source.alias == data_source_alias
+        end
```

Comment:
What return if id or alias not found, throw exception or return undefined? We should mention in the api doc.

Reply:
Done update api doc. Throw exception if the id or alias are not found.